samples/MASMessagingSample/src/main/java/com/ca/mas/masmessagingsample/activity/GroupListActivity.java

```diff
@@ -47,11 +61,41 @@ protected void onCreate(Bundle savedInstanceState) {
         toolbar.setTitle(getTitle());
 
         mContext = this;
+        _freshLaunch = true;
+        mProgress = new ProgressDialog(this);
+        mProgress.show();
         mRecyclerView = (RecyclerView) findViewById(R.id.group_list);
         assert mRecyclerView != null;
 
+        FloatingActionButton searchGroup = (FloatingActionButton) findViewById(R.id.search_group);
+        searchGroup.setOnClickListener(new View.OnClickListener() {
+            @Override
+            public void onClick(View view) {
+                Intent intent = new Intent(mContext, SearchGroupActivity.class);
+                startActivity(intent);
+            }
+        });
+
+        FloatingActionButton createGroup = (FloatingActionButton) findViewById(R.id.create_group);
+        createGroup.setOnClickListener(getGroupListener(false));
+
+        FloatingActionButton deleteGroup = (FloatingActionButton) findViewById(R.id.delete_group);
+        deleteGroup.setOnClickListener(getGroupListener(true));
+
         MAS.start(this, true);
-        MASUser.login("username", "password", getUserCallback());
+        //MASUser.login("manu", "dost1234".toCharArray(), getUserCallback());
+        MASUser.login(getUserCallback());
+
+    }
+
+    @Override
+    public void onResume() {
+        super.onResume();
+        if (!_freshLaunch && (MASUser.getCurrentUser() == null || !MASUser.getCurrentUser().isAuthenticated())) {
```

Comment:
Can we extract MASUser.getCurrentUser() into its own variable here and reuse it?

MASUser user = MASUser.getCurrentUser();
if (!_freshLaunch && (user == null || !user.isAuthenticated()) {